src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -847,4 +847,51 @@ private void testNullHeadGetCigarBAM() {
 
         testNullHeaderCigar(bamRec);
     }
+
+    @Test
+    public void testSetHeaderStrictValid() {
+        SAMRecord sam = createTestRecordHelper();
+        final SAMFileHeader samHeader = sam.getHeader();
```

Comment:
Add a second `testSetHeaderStrictValid` test case that uses `setHeaderStrict()` to set the header on a read to a header that is compatible, but different from the original header.